.circleci/config.yml

```diff
@@ -99,30 +103,83 @@ commands:
+                      elif [[ "$CIRCLE_BRANCH" =~ ^pull/[0-9]+$ ]]; then
+                          PR_NUMBER=$(echo $CIRCLE_BRANCH | sed 's/pull\///')
+                          echo "Deploying VRT for forked PR #$PR_NUMBER"
+                      elif [ -n "$GITHUB_TOKEN" ]; then
+                          # GitHub API fallback
+                          PR_NUMBER=$(curl -s -H "Authorization: Bearer $GITHUB_TOKEN" \
+                              -H "Accept: application/vnd.github+json" \
+                              "https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls?head=$CIRCLE_PROJECT_USERNAME:$CIRCLE_BRANCH&state=open" \
+                              | grep -o '"number":[0-9]*' | head -1 | sed 's/"number"://')
+                          
+                          if [ -n "$PR_NUMBER" ]; then
+                              echo "Deploying VRT for PR #$PR_NUMBER (via GitHub API)"
                           else
-                              attempt=$((attempt + 1))
-                              echo "Deploy failed, retrying ($attempt/$max_attempts)..."
-                              sleep 10  # Adjust delay if needed
+                              echo "⏭️  No PR found - skipping VRT deployment"
+                              exit 0
                           fi
-                      done
+                      else
+                          echo "⏭️  No PR detected - skipping VRT deployment"
+                          exit 0
+                      fi
+
+                      # Only proceed if we have a valid PR number
+                      if [ -z "$PR_NUMBER" ]; then
+                          echo "⏭️  No PR number available - skipping VRT deployment"
+                          exit 0
+                      fi
+
+                      # Create PR hash using just PR number for overwriting deployments
+                      PR_HASH="pr-${PR_NUMBER}"
+
+                      # VRT combination for subdirectory
+                      combination="<< parameters.regression_system >>-<< parameters.regression_color >>-<< parameters.regression_scale >>-<< parameters.regression_dir >>"
+
+                      # Prepare files for deployment
+                      cp projects/documentation/content/favicon.ico test/visual/ || echo "Failed to copy favicon"
+
+                      # Check if AZURE_STORAGE_SAS_TOKEN is set
+                      if [ -z "${AZURE_STORAGE_SAS_TOKEN}" ]; then
+                          echo "ERROR: AZURE_STORAGE_SAS_TOKEN environment variable is not set"
+                          echo "This token is required for Azure Blob Storage deployment"
+                          exit 1
+                      fi
+
+                      # Disable debug mode to prevent token leakage
+                      set +x
+
+                      # Clean the SAS token (remove any whitespace/newlines)
+                      CLEAN_SAS_TOKEN=$(echo "${AZURE_STORAGE_SAS_TOKEN}" | tr -d '\n\r\t ')
```

Comment:
Like the comment below, let’s please switch to Azure AD-based auth here as well. SAS tokens are bearer tokens - anyone with the URL and token can access the blob, so they’re more risky to use in CI environments.